doc/evaluation_design.md

```diff
@@ -0,0 +1,64 @@
+
+Questions to discuss:
+* None of training checkpoints will be evaluted in above two solutions which is different from tf.estimator. tf.estimator will load the lastest checkpoint saved by train to evaluate in a separated pod named evaluator. 
+* Should master preserve the model to checkpoint after evaluation?
+
+## Evaluate by Loading Checkpoint to Pserver
+Generally, deep learning framework preserves model as checkpoint every many seconds or steps. So pserver can load model from checkpoint for evaluation. There are also two designs for pserver to load model. 
+
+1. The master launches additional pods as evalution pserver.\
+When an evaluation job starts, the master will launch additional pods as evaluation pservers and evaluation pservers will load model from checkpoint.\
+The worker gets model from evaluation pservers to inference outputs for evluation tasks and reports outputs to master.\
+The master will release those pods after receiving all outputs of evalution tasks from worker.\
+In this case, evaluation is separated from train. So, some workers can process evaluation tasks and other workers can process train tasks at same time. But more additional pods are needed.
+
+2. Existing pservers stop train task and load model from checkpoint to evaluate.\
+When an evaluation job starts, pservers preserve the current model to temporary file and load model from the lastest checkpoint. After evaluation, pservers will restore model from temporary file preserved and continue to train. \
+The design of pserver will be complicated in this case. What's more,some workers may be reporting gradients when the master change model to evaluation for psersers and pservers can not update model by those gradients at this moment.
+
+
+## Introduction to tf.estimator to Evaluate
+tf.estimator will launch pods as pservers, workers and evalutor when a distributed train job is submitted with ParameterServerStrategy. The evaluator does not participate in train. tf.estimator decides when to evaluate by [throttle_secs](https://www.tensorflow.org/api_docs/python/tf/estimator/EvalSpec#throttle_secs). tf.estimator will check whether there are new checkpoints in checkpoint directory every throttle_secs after the last evaluation ends. The evaluator only has one instance and it will restore the whole model from the lastest checkpoint. So, evaluaion of tf.estimator is not distributed and not fault-tolerant.
+
+## Summary:
+1. If difference with gradients of several mini-batches of model between workers to evaluate is acceptable, the first solution in `Evaluate with Current Model in Pserver` can be adopted. Otherwise，the master should send stop signal to pservers to stop update model after an evaluation job starts.
```

Comment:
It looks like this section is a summary for the different design sections. Perhaps group those design sections into an upper level section so the hierarchies for this markdown is clearer